### ISSUE TITLE
KeyError: 'token_type_ids'

### ISSUE THREAD

**Issue by Qingkongji** [lyuqin/HydraNet-WikiSQL#1]
When running `wikisql_prediction.py` with trained model, I got the error `KeyError: 'token_type_ids'`. The details are as follows.
```
Traceback (most recent call last):
  File "/home/HydraNet-WikiSQL-master/wikisql_prediction.py", line 61, in <module>
    pred_data = SQLDataset(in_file, config, featurizer, False)
  File "/home/HydraNet-WikiSQL-master/featurizer.py", line 262, in __init__
    self.input_features, self.model_inputs, self.pos = self.featurizer.load_data(data_paths, config, include_label)
  File "/home/HydraNet-WikiSQL-master/featurizer.py", line 221, in load_data
    input_feature = self.get_input_feature(example, config)
  File "/home/HydraNet-WikiSQL-master/featurizer.py", line 134, in get_input_feature
    segment_ids = tokenize_result["token_type_ids"]
  File "/home/anaconda3/lib/python3.7/site-packages/transformers/tokenization_utils_base.py", line 273, in __getitem__
    return self.data[item]
KeyError: 'token_type_ids'
```
It seems that the output of the function `tokenizer.encode_plus ` in line 116 of `featurizer.py` doesn't has the key `token_type_ids`.

Please share if there is any solution to this issue.

Thanks and Regards

**Comment by lyuqin** [comment]
Please check if installed library version satisfy transformers=2.3.0 (which is specified in requirements.txt)

**Comment by Qingkongji** [comment]
Thanks for your reply! I used the wrong version 4.0 and I will try the version 2.3.0.

**Comment by Qingkongji** [comment]
It works now. Thanks!

**Comment by Qingkongji** [comment]
@lyuqin Sorry to disturb you again. Now I have the problem: `Process finished with exit code 139 (interrupted by signal 11: SIGSEGV)`. I tried creating a new conda environment to fix it but failed. I positioned the problem in `create_model(config, is_train=False)` by debugging.

Now I use torch=1.3.1 and run the wikisql_prediction.py on GTX 1080 Ti 11G.

**Comment by lyuqin** [comment]
We've never seen this error before. "create_model" is only a wrapper function for Huggingface API, so I guess there is some problem with transformer library in your local env.

Btw, if you are using the trained HydraNet model to do prediction, please run with multi-gpu as the model is trained with multi-gpu and has extra "module." prefix in the weight name. If run with one gpu or cpu, you could see "Missing Key(s) in state_dict" error. The fix is in, e.g., https://discuss.pytorch.org/t/missing-keys-unexpected-keys-in-state-dict-when-loading-self-trained-model/22379/2

**Comment by Qingkongji** [comment]
Thanks for your reminding. I will try to fix it.

**Comment by Qingkongji** [comment]
The problem is solved when I use torch 1.7.1.
